openedx/core/djangoapps/content_tagging/api.py

```diff
@@ -175,6 +190,118 @@ def tag_content_object(
     return get_content_tags(object_key, taxonomy_id=taxonomy.id)
 
 
+def get_object_tree_with_objecttags(
+    content_key: ContentKey,
+    include_children: bool = True,
+) -> tuple[TaggedContent, TaxonomyDict]:
+    """
+    Returns the object with the tags associated with it. If include_children is True, then it will also include
+    the children of the object and their tags.
+    """
+
+    def _get_object_tags(content_key: ContentKey, include_children: bool) -> QuerySet[ObjectTag]:
+        """
+        Return the tags for the object and its children using a single db query.
+        """
+        content_key_str = str(content_key)
+        if not include_children:
+            return ObjectTag.objects.filter(
+                object_id=content_key_str,
+                tag__isnull=False,
+                tag__taxonomy__isnull=False,
+            ).select_related("tag__taxonomy")
```

Comment:
Also I don't think we need to support `include_children=False` or `UsageKey` here - the only use cases we have are "get all blocks from a course" and "get all blocks from a library". We don't need to support "get all children of this specific XBlock in a course". I think you can shorten this function a lot by removing those.

Reply:
Done: db9116d906586417ee9fae9929d32551afcd79a6